sdk/storage/azure-storage-file-share/azure/storage/fileshare/_shared_access_signature.py

```diff
@@ -127,11 +128,12 @@ def generate_file(
         :type sts_hook: Optional[Callable[[str], None]]
         :returns: The generated SAS token for the account.
         :rtype: str
-        '''
+        """
         resource_path = share_name
         if directory_name is not None:
-            resource_path += '/' + _str(directory_name) if directory_name is not None else None
```

Comment:
Wanted to point out this line specifically - `mypy` doesn't like the `None` string addition operations. To make the code more concise and readable, a single `if` `None` check is sufficient.

Reply:
Looks good